### ISSUE TITLE
Lesson design

### ISSUE THREAD

**Issue by dafnevk** [carpentries-incubator/deep-learning-intro#49]
## Target audience
The main audience of this carpentry lesson is PhD students that have little to no experience with deep learning. 

### Pain points of target audience
Probably have overhyped expectations of deep learning. They don’t know if it’s the right tool for their situations. They have no idea what it takes to actually do deep learning. 

Want to quickly have some useful skills for their own data. 

## Required Pre-Knowledge
-  **Basic Machine Learning Knowledge** – Data Cleaning, Train & Test Split, Overfitting & Underfitting, Metrics (Accuracy, Recall, etc.),  
- **Python** – Previous Experience programming in python is required (Refer to Python Data Carpentry Lesson )
- **Pandas** – Knowledge of the Pandas python package 

## Learning objectives
- **Prepare input data for use for deep learning** – This includes cleaning data, filling missing values, normalizing, and transforming categorical columns into dummy encoding. 

- **Design and train a Deep Neural Network** – This includes knowledge of when to different types of layers 

- **Troubleshoot the learning process** – Often when designing neural networks training will not automatically work very well. This requires setting the parameters of the training algorithm correctly, modifying the design of the network or changing the data pre-processing 

- **Measure the performance of the network** – After training the performance of the network should be checked to prevent overfitting. 

- **Visualizing Data and Results** – Within each episode how to visualize data and results 

- **Re-use existing network architectures with and without pre-trained weights** – Re-use of architectures is common in deep learning. Especially when using pre-trained weights (transfer-learning) it can also be very powerful. 

## Episodes
- Episode 1: Introduction to Deep Learning #42 
- Episode 2:  Introduction to Keras #43 
- Episode 3: monitoring the training process #44 
- Episode 4: Networks are like Onions #45 
- Episode 5 (optional): Transfer Learning  #46 
- Episode 6 (optional) Data Augmentation #47 
- Episode 7: Outlook #48

**Comment by HaoZeke** [comment]
One question/comment is regarding the installation setup. Do we expect the learners to setup the libraries locally or use Colab? Generally we prefer local installs in most of the Carpentries content, but I feel like for an intro to DL, Colab is a lot gentler (also with better resources than many learners might have).

**Comment by HaoZeke** [comment]
I also noticed that the main focus is image data, should there be a section on NLP?

**Comment by dafnevk** [comment]
Thanks @HaoZeke for these questions, we actually discussed both these points.
We thought it would be good and possible to have leaners install everything locally, and we will make sure to use small datasets so that everything runs on a laptop. However, it would be good to make a note in the last episode that often you will encounter situations where your laptop is not good enough and you would like to use a machine with a GPU, or a compute cluster. There we can refer to other courses or reading material.

Regarding NLP: We thought that would be too much out of scope, since the way language is handled in DL is quite particular, whereas the techniques you learn for vision are generically applicable to many types of data.

**Comment by Sabryr** [comment]
1. Suggestion for "Required Pre-Knowledge"  - Not so much focus on the data science aspects and provide the data cleaning etc.. as part of the course.  Then make the learner profile more explicit, i.e. This course is for you if you in one of the following groups: "1. I have little or no deep learning background but I know statistics, linear algebra .. 2. "
2. Fort the library requirements, I could volunteer to create a conda installation file with everything needed, if others think that this  is a good solution. (to be used, from Unix, Windows and even from a Jupyter )

**Comment by svenvanderburg** [comment]
I really like the lesson design so far! It's practical and coherent.

If I can make 1 suggestion:
* With the current setup, all I did in this course was play around with toy networks. I would still not really know what it takes to actually apply deep learning in the real world. This is a shame because it is actually really easy nowadays to apply state-of-the-art networks. I would vote for making episode 5 required and learn how to use a pre-trained state-of-the-art network.

**Comment by psteinb** [comment]
I'd like to provide some feedback to the lesson design. However, as it is currently an issue, I am not sure how to do that - besides a comment. 

It might make sense, to adapt the lesson design ideas into a markdown file under `_extras`.

If people are fine with this, I'd start by submitting a PR that transforms this issue into `extras/design.md` for example. This way, feedback and suggestions can be channeled to specific aspects of the design like only the assumptions on the learners, or only the learning objectives, or ... you name it.

**Comment by dafnevk** [comment]
Sounds like an excellent idea, go for it!

**Comment by tobyhodges** [comment]
Perhaps you have already seen and considered this, but the lesson would be a good fit for the Lesson Development Study Groups program we will launch next month. I would love to see an application from one or more of you who will be leading the development of this lesson.

**Comment by psteinb** [comment]
@tobyhodges thank you for pointing "us" to [1]. I think this is a good initiative. 

From my excursions to 'teaching teach together' by Greg Wilson, I'd be interested to participate. Besides, I might have to teach Deep Learning in March/April time frame anyhow. Let me consider it.

However, for making a difference to the repo, it be great if someone with merge permission for this repo could embark on this adventure with me. That might boost this activity even more so.

[1]: https://carpentries.org/blog/2020/12/lesson-development-study-groups/

**Comment by dafnevk** [comment]
I'm closing this issue, because the design is now put into a document within the lessons (https://github.com/carpentries-incubator/deep-learning-intro/pull/56). Suggestions for the design can be made by making a pull request on the file `_extras/design.md`
